File: 867. Transpose Matrix/867.go
package main

import (
	"fmt"
)

func main(){
	fmt.Println(transpose([][]int{{1,2,3},{4,5,6}}))
}

func transpose(A [][]int) [][]int {
	var B [][]int
    for collength:=0; collength<len(A[0]); collength++{
		var tmplist []int
		for rowlength:=0; rowlength<len(A); rowlength++{
			tmplist = append(tmplist, A[rowlength][collength])
		}
		B = append(B, tmplist)
	}
	return B
}

File: 867. Transpose Matrix/867.py
class Solution:
    def transpose(self, A):
        B = list()
        for i in range(len(A[0])):
            tmplist = list()
            for j in range(len(A)):
                tmplist.append(A[j][i])
            B.append(tmplist)

        return B
    
    def transpose1(self, A):
        return list(zip(*A))

a = Solution()
print(a.transpose1([[1,2,3],[4,5,6]]))

